parcels/field.py

```diff
@@ -625,20 +628,21 @@ def search_indices_curvilinear(self, x, y, z, xi, yi, ti=-1, time=-1, search2D=F
             it += 1
             if it > maxIterSearch:
                 print('Correct cell not found after %d iterations' % maxIterSearch)
-                raise FieldSamplingError(x, y, 0, field=self)
+                raise FieldOutOfBoundError(x, y, 0, field=self)
 
         if grid.zdim > 1 and not search2D:
             if grid.gtype == GridCode.CurvilinearZGrid:
                 (zi, zeta) = self.search_indices_vertical_z(z)
+                if zi == -999:
```

Comment:
Why do we need this -999 here? Is it because we don't know `x` and `y` in `self.search_indices_vertical_z`? 
If that's the case, would it not be cleaner to add `x` and `y` as arguments to that function too, so we can throw the error there?
Or, alternatively and perhaps more correctly coded, throw a `VerticalSamplingError` in `self.search_indices_vertical_z`, and catch that error here, then throw the `FieldOutOfBoundError`.